frontend/src/pages/RegisterPage.js

```diff
@@ -2,14 +2,63 @@ import Head from '../components/Head';
 import Navlink from '../components/Navlink';
 import Footer from '../components/Footer';
 import FormTemplate from '../components/FormTemplate';
-import { 
-  Form,
-  Container,
-  Row,
-  Button
-} from 'react-bootstrap';
+import Swal from 'sweetalert2';
+import { useState, useEffect } from 'react';
+import { Form, Container, Row, Button } from 'react-bootstrap';
 
 export default function RegisterPage() {
+  const [ name, setName ] = useState('');
+  const [ username, setUserName ] = useState('');
+  const [ email, setEmail ] = useState('');
+  const [ password, setPassword ] = useState('');
+  const [ passwordConfirm, setPasswordConfirm ] = useState('');
+  const [ btnActive, setBtnActive ] = useState(false);
+
+  // set submit to active if criteria is met.
+  useEffect(() => {
+    if (((name && username && email && password && passwordConfirm) !== "") && (password === passwordConfirm)) {
+      setBtnActive(true)
+    } else {
+      setBtnActive(false)
+    }
+  }, [ name, username, email, password, passwordConfirm ]);
+
+  // register a user
+  function registerUser(e) {
+    e.preventDefault();
+
+    fetch('http://localhost:8000/api/register', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Requested-With': 'XMLHttpRequest'
+      },
+      body: JSON.stringify({
+        name: name,
+        username: username,
+        email: email,
+        password: password,
+        password_confirmation: passwordConfirm
+      })
+    }).then(res => res.json()).then(data => {
```

Comment:
Hmm I just realize we can also update this one. Usually we seperate Api function to different file so we can reuse them if needed and to avoid making component code longer and difficult to trace.

so as example we have here `register` api, then i think you next feature is `login` and `logout` so we can put those api calls in a different file like `/api/UserAuthApi` then have it something like this 
```
const UserAuthApi = {
  register: (params) => {
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'X-Requested-With': 'XMLHttpRequest'
      },
      body: JSON.stringify({
        name: params.name,
        username: params.username,
        email: params.email,
        password: params.password,
        password_confirmation: params.passwordConfirm
      })
    }

    return fetch('http://localhost:8000/api/register', options);
  },
  login: (params) => {
    // code here
  },
  logout: (params) => {
    // code here
  },
};

export default UserAuthApi;
```

This way well be able to separate the concern for sending request and handling response like this
```
// import
import UserAuthApi from '../api/UserAuthApi';


// call method
UserAuthApi.register({
  name: name,
        username: username,
        email: email,
        password: password,
        password_confirmation: passwordConfirm
}).then(res => {
  // handle succcess
}).catch(err => {
  // handle error
});

```

Reply:
This is just exmaple you can make it cleaner how ever you feel like :100: